### PR TITLE
Rewrite documentation

### LINKED ISSUE #3: New documentation
* ~~convert to reST~~
* review current functionality and make sure it is all documented
* add a quickstart section with some easy examples.

### PR DESCRIPTION
**Pull request by ddeboer**
View built version at http://portphp.readthedocs.org/en/docs/.

Note: this is WIP; there’s much that still needs to be done.

Fix #3.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Port Documentation
+==================
+
+This is the documentation for [PortPHP](https://github.com/portphp/portphp).
+
+PortPHP is a data import/export pipeline:
```
Comment (sagikazarmark) [inline docs/index.md]: Hm, sounds familiar...maybe from plumphp? :stuck_out_tongue_winking_eye:

File: docs/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+## Standalone reader and writer
+
+If you want to do a quick, one-off import, you can use the [readers](readers.md)
+and [writers](writers.md) on their own. Just create a reader and a writer and
+iterate over the reader:
+
+```php
+use Port\Reader\Csv;
```
Comment (sagikazarmark) [inline docs/quickstart.md]: Isn't it `Port\Csv\CsvReader`?

File: docs/index.md (hunk 1 above)
Comment (ddeboer) [inline docs/index.md]: Haha yeah, pipeline, ships, ports, makes sense, doesn’t it. :wink:

Comment (sagikazarmark) [inline docs/index.md]: It does, I am not complaining. :wink:

File: docs/readers.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    'SELECT u.id, u.username, g.name FROM `user` u INNER JOIN groups g ON u.group_id = g.id'
+);
+```
+
+## Doctrine ORM/ODM
+
+Reads data through the [Doctrine ORM](http://www.doctrine-project.org/projects/orm.html):
+
+```php
+use Port\Reader\Doctrine\DoctrineReader;
```
Comment (Baachi) [inline docs/readers.md]: Maybe `Doctrine\ORMReader` is better?

File: docs/readers.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+$reader = new DoctrineReader($entityManager, 'Your\Namespace\Entity\User');
+```
+
+## Excel
+
+Install the Excel add-on:
```
Comment (Baachi) [inline docs/readers.md]: `add-on` vs `extension`?

File: docs/readers.md (hunk 3 above)
Comment (ddeboer) [inline docs/readers.md]: But it accepts ODM as well as it only depends on the general `Doctrine\Common\Persistence\ObjectManager` interface.

File: docs/readers.md (hunk 4 above)
Comment (ddeboer) [inline docs/readers.md]: Yes, we need to standardise naming. See also #8.

File: docs/readers.md (hunk 3 above)
Comment (sagikazarmark) [inline docs/readers.md]: Yup, ORM is not appropriate here.

Comment (Baachi) [inline docs/readers.md]: Nope it doesn't accept ODM

Comment (sagikazarmark) [inline docs/readers.md]: Yet. But there was a PR in the original package, which makes sense IMO. (The one with the PHPUnit problem)

Comment (ddeboer) [inline docs/readers.md]: This one: https://github.com/ddeboer/data-import/pull/83.

Comment (sagikazarmark) [inline docs/readers.md]: I am pretty sure it is almost impossible to rebase that branch so it can work. :wink:

Comment (ddeboer) [inline docs/readers.md]: Sure, but there’s always copy/paste. :stuck_out_tongue_closed_eyes:

Comment (sagikazarmark) [inline docs/readers.md]: Yeah :)

Comment (Baachi) [inline docs/readers.md]: I would rather having a ORMReader and a ODMReader. Its more readable and mades the code less complex.

Comment (ddeboer) [inline docs/readers.md]: Please continue discussion in #13.

**Comment by ddeboer** [comment]
Fix #38.

**Pull request merged**